dask/dataframe/io/csv.py

```diff
@@ -941,7 +941,8 @@ def to_csv(
     if single_file:
         first_file = open_file(filename, mode=mode, **file_options)
         value = to_csv_chunk(dfs[0], first_file, **kwargs)
-        append_mode = mode.replace("w", "") + "a"
+        append_mode = mode if mode.endswith("a") else mode + "a"
+        append_mode = append_mode.replace("w", "")
```

Comment:
If I understand the file modes correctly, we also want to remove `x` (_open for exclusive creation, failing if the file already exists_) from the mode string.

Reply:
This problem has not been introduced in this PR, so we can also push this into a dedicated issue.

Reply:
I'm happy to take a look at this either under this PR or as a separate one - whichever is easiest? I think adding `x` might need some extra logic /thought though.

The user can select a write mode as a keyword argument (which defaults to "w"). When this function receives a separate keyword for "single_file", it adds "a" onto the write mode so that the partitions don't overwrite each other.

The issue at the moment, is that if the user has already said they want to use append mode, then the code is overcompensating, adding an "a" anyway and writing in something like "aa" mode which is invalid.

If a user adds "x" into the mode, then they probably want some guarantees that the file isn't being overwritten. The function I guess needs to make sure it does something like write the first request in "x" mode, and then switch out so that:
- The write will fail in the file already exists
- Dask won't be blocked from using append to write multiple partitions into a single file

@hendrikmakait, does that logic check out with you?

Reply:
@benrutter: I think I'd prefer to move this to a dedicated issue. I don't want to block your PR by this. It seems a couple of folks would be happy to see it in the next release. (For context: I wrote the comment before realizing you hadn't introduced this logic.) Generally, I think your logic checks out. I'll create a new issue with a small reproducer and cc you on that one. Sounds good?